docs/source/sci_api_tutorial/armer.rst

```diff
@@ -2,21 +2,297 @@
+For specific details, please refer to the `original paper`linked (https://doi.org/10.1021/acs.jcim.3c00618). 
 
-API Config Dict
+.. dropdown:: :fa:`eye,mr-1` Click to learn more about ARMer's architechture
+
+    The ARMer Python library consists of two classes: the job class and the HPC class. 
+    
+    The job class (called ClusterJob in the code) defines properties and functions that are associated with job configuration, submission, and dynamic monitoring of job completion. 
+    
+    The HPC class (subclasses of ClusterInterface in the code) supports the job class with properties and functions to mediate shell input/output in the user’s local HPC where ARMer is deployed. The HPC class files are stored in a folder named “cluster”. In the folder, _interface.py defines an abstract HPC class as the code interface and accre.py defines an example concrete HPC class we made for our local HPC at Vanderbilt. Users can create new files under this folder defining new concrete HPC classes to easily modify ARMer Python library to be compatible with their local HPC cluster. The instances of the HPC class are used as input for generating the Job instance. The methods of the HPC class are used by the Job instance through the HPC instance to interface with the corresponding local HPC cluster. The new HPC class user defines is required to fulfill the code interfaced defined by the abstract HPC class in _interface.py to make sure they are compatible with the Job class. It is enforced by requiring (by the Job class) all HPC classes to inherit the abstract HPC class so that the new HPC class has to define some required methods (otherwise python will raise an error). 
+    
+    Then, the ARMer library enables the “workflow script” to run shell commands on other computational nodes these commands are wrapped in the job scripts in the HPC clusters.
+
+.. dropdown:: :fa:`eye,mr-1` Click to learn more about ARMer's workflow
+
+    The workflow of ARMer
+    ==============================================
+
+    .. panels::
+
+        :column: col-lg-12 col-md-12 col-sm-12 col-xs-12 p-2 text-left
+
+        .. image:: ../../figures/armer_architect.png
+            :width: 100%
+            :alt: single_point_api_io                  
+        |
+
+
+
+Three Main Functions of ARMer
 ==============================================
-| This section describe a soft-enforced standard format of 
-  the configuration dictionary used as input of all the 
-  Science APIs.
-| For example: 
-| ``quantum.single_point()`` needs 
-  to know what resource/account it needs to use when running
-  commands through ARMer. We have a ``cluster_job_config``
-  argument in this function as a dictionary obtain this information.
-  The format of this dictionary is soft-enforced.
-
-TODO
-  
+The ARMer tool contains three main functions: config job (i.e.: submission script, etc.), submit, monitor (i.e.: wait_to_end).
+
+1. Job Configuration
```

Comment:
Let's have the function name in the title
For example this one can be
`config_job` for job configuration

Reply:
Make sure the function name is in the code style

Reply:
Also it will be great if you can make DFD for each of the 3 functions